static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx

```diff
@@ -0,0 +1,131 @@
+import styled from '@emotion/styled';
+import {Location, LocationDescriptorObject} from 'history';
+
+import SortLink from 'sentry/components/gridEditable/sortLink';
+import Tooltip from 'sentry/components/tooltip';
+import Truncate from 'sentry/components/truncate';
+import {Organization, PageFilters} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {getIssueFieldRenderer} from 'sentry/utils/dashboards/issueFieldRenderers';
+import {TableDataRow, TableDataWithTitle} from 'sentry/utils/discover/discoverQuery';
+import {isFieldSortable} from 'sentry/utils/discover/eventView';
+import {getFieldRenderer} from 'sentry/utils/discover/fieldRenderers';
+import {
+  fieldAlignment,
+  getAggregateAlias,
+  getEquationAliasIndex,
+  isEquationAlias,
+} from 'sentry/utils/discover/fields';
+import {DisplayType, Widget, WidgetType} from 'sentry/views/dashboardsV2/types';
+import {eventViewFromWidget} from 'sentry/views/dashboardsV2/utils';
+import {ISSUE_FIELDS} from 'sentry/views/dashboardsV2/widgetBuilder/issueWidget/fields';
+import TopResultsIndicator from 'sentry/views/eventsV2/table/topResultsIndicator';
+import {TableColumn} from 'sentry/views/eventsV2/table/types';
+
+// Dashboards only supports top 5 for now
+const DEFAULT_NUM_TOP_EVENTS = 5;
+
+type Props = {
+  location: Location;
+  organization: Organization;
+  selection: PageFilters;
+  widget: Widget;
+  tableData?: TableDataWithTitle;
+};
+
+export const renderGridHeaderCell =
+  ({selection, widget, tableData}: Props) =>
+  (column: TableColumn<keyof TableDataRow>, _columnIndex: number): React.ReactNode => {
+    const eventView = eventViewFromWidget(
+      widget.title,
+      widget.queries[0],
+      selection,
+      widget.displayType
+    );
+    const tableMeta = tableData?.meta;
+    const align = fieldAlignment(column.name, column.type, tableMeta);
+    const field = {field: column.name, width: column.width};
+    function generateSortLink(): LocationDescriptorObject | undefined {
+      // TODO: need write sort link generation for widget viewer
+      return undefined;
+    }
+    const currentSort = eventView.sortForField(field, tableMeta);
+    const canSort = isFieldSortable(field, tableMeta);
+    const titleText = isEquationAlias(column.name)
+      ? eventView.getEquations()[getEquationAliasIndex(column.name)]
+      : column.name;
+
+    const title = (
+      <StyledTooltip title={titleText}>
+        <Truncate value={titleText} maxLength={60} expandable={false} />
+      </StyledTooltip>
+    );
+
+    return (
+      <SortLink
+        align={align}
+        title={title}
+        direction={currentSort ? currentSort.kind : undefined}
+        canSort={canSort}
+        generateSortLink={generateSortLink}
+      />
+    );
+  };
+
+export const renderGridBodyCell =
+  ({location, organization, widget, tableData}: Props) =>
+  (
+    column: TableColumn<keyof TableDataRow>,
+    dataRow: TableDataRow,
+    rowIndex: number,
+    columnIndex: number
```

Comment:
shouldn't we use the following types?

```suggestion
   column: GridColumnOrder<ObjectKey>,
    dataRow: Record<string, any>,
    rowIndex: number,
    columnIndex: number
```

Reply:
👍 updated the type for `renderGridBodyCell`